File: Homework/JavaScript/#1 - Fundamentals/#2 - Conditional Statements/6.QuadraticEquation/QuadraticEquation.js
// Write a script that enters the coefficients a, b and c of a quadratic equation a*x2 + b*x + c = 0 and calculates and prints its real roots. Note that quadratic equations may have 0, 1 or 2 real roots.

function EnterNumber() {
    var a = parseFloat(document.getElementById("a").value);
    var b = parseFloat(document.getElementById("b").value);
    var c = parseFloat(document.getElementById("c").value);

    CalculateQuadraticEquation(a, b, c);
}

function CalculateQuadraticEquation(a, b, c) {
    d = b * b + 4 * a * c;
    document.getElementById("d").value = "Discriminant = " + d.toFixed(2);
    if (d > 0) {
        var xOne = (-b + Math.sqrt(d)) / 2 * a;
        var xTwo = (-b - Math.sqrt(d)) / 2 * a;
        document.getElementById("x-one").value = "X1 = " + xOne.toFixed(2);
        document.getElementById("x-two").value = "X2 = " + xTwo.toFixed(2);
    }
    else if (d = 0) {
        var x = -b / 2 * a;
        document.getElementById("x-one").value = "X1 = X2 = " + x.toFixed(2);
    }
    else {
        document.getElementById("x-one").value = "Equation has no real roots!";
    }
}

File: Homework/C#/Part I/#5 - ConditionalStatements/6.QuadraticEquation/QuadraticEquation.cs
// Write a program that enters the coefficients a, b and c of a quadratic equation. a*x2 + b*x + c = 0. and calculates and prints its real roots. Note that quadratic equations may have 0, 1 or 2 real roots.

using System;

namespace QuadraticEquation
{
    class QuadraticEquation
    {
        static void Main()
        {
            double a;
            double b;
            double c;
            double d;
            double numOne;
            double numTwo;
            double den;
            double xOne;
            double xTwo;

            Console.Write("Please enter a number of 'a': ");
            a = double.Parse(Console.ReadLine());

            Console.Write("Please enter a number of 'b': ");
            b = double.Parse(Console.ReadLine());

            Console.Write("Please enter a number of 'c': ");
            c = double.Parse(Console.ReadLine());

            d = b * b - 4 * a * c;
            if (d > 0)
            {
                numOne = -b + Math.Sqrt(d);
                numTwo = -b - Math.Sqrt(d);
                den = 2 * c;
                xOne = numOne / den;
                xTwo = numTwo / den;
                Console.WriteLine("D  = {0:N2}\nX1 = {1:N2}\nX2 = {2:N2}", d, xOne, xTwo);
            }
            else if (d == 0)
            {
                numOne = -b;
                den = 2 * c;
                xOne = numOne / den;
                Console.WriteLine("D = 0\nX1 = X2 = {0:N2}", xOne);
            }
            else
            {
                Console.WriteLine("Квадратното уравнение няма реални корени");
            }
        }
    }
}
